SparseGrids/tasgridExternalTests.cpp

```diff
@@ -1311,6 +1311,28 @@ bool ExternalTester::testAllRefinement() const{
         if (!testAnisotropicRefinement(f, &grid, type_ipcurved, 1, np, errs, 7)){
             cout << "ERROR: failed anisotropic refinement using leja ipcurved nodes for " << f->getDescription() << endl;  pass2 = false;
         }
+    }{
+        const BaseFunction *f = &f21c1c2periodic;
+        grid.makeFourierGrid(f->getNumInputs(), f->getNumOutputs(), 3, type_hyperbolic);
+        double transform_a[2] = {-1.0, -1.0};
+        double transform_b[2] = { 1.0,  1.0};
+        grid.setDomainTransform(transform_a, transform_b);
+        int np[5] = { 17, 35, 111, 273, 759 };
+        double errs[5] = { 1.28e-2, 2.80e-3, 1.97e-4, 6.78e-5, 5.65e-5 };
+        if (!testAnisotropicRefinement(f, &grid, type_hyperbolic, 1, np, errs, 5)){
+            cout << "ERROR: failed anisotropic refinement using Fourier hyperbolic nodes for " << f->getDescription() << endl;  pass2 = false;
+        }
+    }{
+        const BaseFunction *f = &f21c1c2periodic;
+        grid.makeFourierGrid(f->getNumInputs(), f->getNumOutputs(), 3, type_level);
+        double transform_a[2] = {-1.0, -1.0};
+        double transform_b[2] = { 1.0,  1.0};
+        grid.setDomainTransform(transform_a, transform_b);
+        int np[5] = { 81, 135, 297, 819, 945 };
+        double errs[5] = { 1.32e-3, 1.90e-4, 6.73e-5, 5.66e-5, 2.19e-6 };
+        if (!testAnisotropicRefinement(f, &grid, type_level, 1, np, errs, 5)){
```

Comment:
Why are you using `type_level` here. The type for the refinement doesn't have to match the type in the constructed grid. What is the meaning of the level refinement?

Reply:
That was a mistake. `estimateAnisotropicCoefficients()` for Fourier grids supports "level refinement" for functions that are analytic in some strip. The derivation of matrix entries comes from the Fourier coefficients decaying exponentially and taking the logarithm of both sides. But that strategy isn't appropriate here since the Fourier coefficients do not decay exponentially. I will change it.